### PR TITLE
Feat :FC VAT Debt Calculation with Breathing Space

### PR DESCRIPTION
**Pull request by thadeusssewanyana**

### CODE REVIEW

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      | 0                    | 400000            |
+    And the 1st fc vat debt summary will contain
+      | debtItemChargeId | interestDueDailyAccrual | interestRate |
+      | 123              | 0                       | 0.0          |
+
+
+  Scenario: Breathing space for no interest bearing debt with payments.
+    Given a fc vat debt item
+      | debtItemChargeId | originalAmount | periodEnd  | interestRequestedTo | interestIndicator |
+      | 123              | 500000         | 2022-04-01 | 2021-01-01          | Y                 |
```
Comment (rogerball) [inline src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: Should interestIndicator be N?

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      | 28                   | 400000            |
+    And the 1st fc vat debt summary will contain
+      | debtItemChargeId | interestDueDailyAccrual | interestRate |
+      | 123              | 28                      | 2.6          |
+
+
+  Scenario: Breathing space for interest bearing debt with no payments.
+    Given a fc vat debt item
+      | debtItemChargeId | originalAmount | periodEnd  | interestRequestedTo | interestIndicator |
+      | 123              | 500000         | 2022-04-01 | 2022-01-04          | Y                 |
```
Comment (rogerball) [inline src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: interestRequestedTo should be before debtRespiteTo?

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      | 37                   | 500000            |
+    And the 1st fc vat debt summary will contain
+      | debtItemChargeId | interestDueDailyAccrual | interestRate |
+      | 123              | 37                      | 2.75         |
+
+
+  Scenario: Non interest bearing debt should not have breathing space applied
+    Given a fc vat debt item
+      | debtItemChargeId | originalAmount | periodEnd  | interestRequestedTo | interestIndicator |
+      | 123              | 500000         | 2022-04-01 | 2023-03-01          | N                 |
```
Comment (rogerball) [inline src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: interestRequestedTo should be before debtRespiteTo?

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature (hunk 1 above)
Comment (thadeusssewanyana) [reply src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: updated

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature (hunk 2 above)
Comment (thadeusssewanyana) [reply src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: updated

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature (hunk 3 above)
Comment (thadeusssewanyana) [reply src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: uodated

File: src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature [hunk 4]
```diff
@@ -0,0 +1,160 @@
+Feature: FC VAT Debt Calculation with Breathing Space
+
+  Scenario: Breathing space for interest bearing debt with payments.
+    Given a fc vat debt item
+      | debtItemChargeId | originalAmount | periodEnd  | interestRequestedTo | interestIndicator |
+      | 123              | 500000         | 2022-04-01 | 2021-01-01          | Y                 |
```
Comment (rogerball) [inline src/test/resources/features/ifs/fieldCollectionsIFS/VATFCBreathingSpace.feature]: interestRequestedTo should be prior to debtRespiteTo?

**Review by rogerball — APPROVED** [review]

**Pull request merged**